api.c

```diff
@@ -348,6 +349,32 @@ int tcmu_emulate_evpd_inquiry(
 		return SAM_STAT_GOOD;
 	}
 	break;
+	case 0x80: /* Unit Serial Number */
+	{
+		char data[512];
+		char *wwn;
+		uint32_t len;
+
+		memset(data, 0, sizeof(data));
+
+		data[1] = 0x80;
+
+		wwn = tcmu_get_wwn(dev);
+		if (!wwn) {
+			return tcmu_set_sense_data(sense, HARDWARE_ERROR,
+						   ASC_INTERNAL_TARGET_FAILURE,
+						   NULL);
+		}
+
+		len = sprintf(&data[4], "%s", wwn);
```

Comment:
Actually in kernel, the unit_serial has been limited to INQUIRY_VPD_SERIAL_LEN, it is 254. So if every thing works correctly, this shouldn't happen.

But as it maybe mysteriouly corrupt the data for some unknown reason(for now it won't happen for the current code), so I will limit it to 256 using snprintf().

Thanks,